File: game/src/Creature.java
public class Creature {
    private String _name;
    private char _symbol;
    private Location _location;
    private Direction _direction = Direction.UP;

    public Creature(String name, char symbol, Location location) {
        this._name = name;
        this._symbol = symbol;
        this._location = location;
    }

    public String getName() {
        return _name;
    }

    public void setName(String name) {
        this._name = name;
    }

    public Location getLocation() {
        return _location;
    }

    public void setLocation(Location location) {
        this._location = location;
    }

    public char getSymbol() {
        return _symbol;
    }

    public void setSymbol(char symbol) {
        this._symbol = symbol;
    }

    public void moveUp() {
        this._location.y--;
    }

    public void moveDown() {
        this._location.y++;
    }

    public void moveLeft() {
        this._location.x--;
    }

    public void moveRight() {
        this._location.x++;
    }

    public void move(Direction direction) {
        this._direction = direction;

        switch (direction) {
            case UP -> this.moveUp();
            case DOWN -> this.moveDown();
            case LEFT -> this.moveLeft();
            case RIGHT -> this.moveRight();
        }
    }

    public void move() {
        this.move(this._direction);
    }

    public Direction getDirection() {
        return _direction;
    }

    public void setDirection(Direction direction) {
        this._direction = direction;
    }

}

File: mangPython/Creature.py
from Direction import Direction


class Creature:
    _symbol: str
    _location: list[int]
    _name: str
    _direction: Direction = Direction.UP

    def __init__(self, name: str, symbol: str, location: list[int]):
        self._symbol = symbol
        self._location = location
        self._name = name

    @property
    def symbol(self) -> str:
        return self._symbol

    @symbol.setter
    def symbol(self, symbol: str):
        self._symbol = symbol

    @property
    def location(self):
        return self._location

    @location.setter
    def location(self, location: list[int]):
        self._location = location

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self, name: str):
        self._name = name

    @property
    def direction(self) -> Direction:
        return self._direction

    @direction.setter
    def direction(self, direction: Direction):
        self._direction = direction

    def moveUp(self):
        self._location[0] -= 1

    def moveDown(self):
        self._location[0] += 1

    def moveLeft(self):
        self._location[1] -= 1

    def moveRight(self):
        self._location[1] += 1

    def __str__(self) -> str:
        return f"{self._symbol} {self._name} {self._location} {self._direction}"
